### PR TITLE
consensus, core, tests: implement Metropolis EIP 649

### PR DESCRIPTION
**Pull request by karalabe**
Implements https://github.com/ethereum/EIPs/pull/669.

The PR also pulls in the latest test suite, which among other throws out explicit log validation to implicit one via RLP(logs).

---

Note, some tests are failing because the consensus test suite was only partially updated to the new log format, so the non-updated ones cannot be parsed. I'll keep checking the test suite to see if there are updates to fix them.

### CODE REVIEW

File: tests/vm_test_util.go [hunk 1]
```diff
@@ -44,14 +44,14 @@ func (t *VMTest) UnmarshalJSON(data []byte) error {
...
-	Exec          vmExec               `json:"exec"`
-	Logs          []stLog              `json:"logs"`
-	GasRemaining  *math.HexOrDecimal64 `json:"gas"`
-	Out           hexutil.Bytes        `json:"out"`
-	Pre           core.GenesisAlloc    `json:"pre"`
-	Post          core.GenesisAlloc    `json:"post"`
-	PostStateRoot common.Hash          `json:"postStateRoot"`
+	Env           stEnv                 `json:"env"`
+	Exec          vmExec                `json:"exec"`
+	Logs          common.UnprefixedHash `json:"logs"`
```
Comment (fjl) [inline tests/vm_test_util.go]: Please remove the logs check so we can merge.

Comment (karalabe) [inline tests/vm_test_util.go]: I'll wait to see how many fail (just updates the test suite), and will disable only the failing tests (so we're not overly zealous and accidentally miss consensus issues).

**Comment by holiman** [comment]
This was already here, not new for this PR: https://github.com/ethereum/go-ethereum/pull/15028/files#diff-e226000964244889df127e5b44dbf192R546 : 

```golang
 		r.Div(blockReward, big32)
 		reward.Add(reward, r)
```
Looks un-intuitive (reusing `r` which has a totally different meaning on the lines before) and unnecessary (always divide blockReward by 32). 
I'd prefer: 
```

	blockReward := frontierBlockReward
        newphewReward := frontierNewphewReward
 	if config.IsMetropolis(header.Number) {
 		blockReward = metropolisBlockReward
                nephewReward:= metropolisNephewReward
 	}
       ...
 		reward.Add(reward, nephewReward)
```

**Comment by devsnd** [comment]
@karalabe I noticed that the `big299999` is 299_000 and not 3_000_000 (-1) as in the EIP

https://github.com/ethereum/go-ethereum/pull/15028/files#diff-e226000964244889df127e5b44dbf192R310
https://github.com/ethereum/go-ethereum/pull/15028/files#diff-e226000964244889df127e5b44dbf192R351

Is this on purpose?

**Comment by holiman** [comment]
@devsnd yes, because he's using `parent.Number` instead of `block.Number`: https://github.com/ethereum/go-ethereum/pull/15028/files#diff-e226000964244889df127e5b44dbf192R355

**Comment by holiman** [comment]
@devsnd OMG wow good catch!

**Comment by karalabe** [comment]
@fjl I've disabled the 3 failing test files, we can readd them later.
@holiman I've fixed the 2_999_999 issue, PLTA.
@devsnd 2,999,999 is correct because the code there is working on the parent block number, not the actual block number. Though do ping me if you still think it's wrong.

**Comment by holiman** [comment]
Just to be clear, my initial comment above was to point out that it was supposed to be one less than 3M, but then I understood that @devsnd had correctly spotted an error - so again, good catch!

**Comment by holiman** [comment]
PTAL:ed, looks good to me

**Pull request merged**